File: java/fizz_buzz/src/test/java/com/devartis/exercise/fizzbuzz/FizzBuzzTest.java
package com.devartis.exercise.fizzbuzz;

import org.junit.Assert;
import org.junit.Test;

import static org.junit.Assert.assertArrayEquals;

public class FizzBuzzTest {

    @Test
    public void shouldReturnFirstThreeElements() {
        assertArrayEquals(
                new String[]{"1", "2", "Fizz"},
                FizzBuzzConverter.convert(3)
        );
    }

    @Test
    public void shouldReturnTheFirst6Elements() {
        assertArrayEquals(
                new String[]{"1", "2", "Fizz", "4", "Buzz", "Fizz"},
                FizzBuzzConverter.convert(6)
        );
    }

    @Test
    public void shouldReturnFizzBuzzFor15() {
        String[] convert = FizzBuzzConverter.convert(15);
        Assert.assertEquals("FizzBuzz", convert[convert.length - 1]);
    }

    @Test
    public void shouldReturn150000Elements() {
        String[] convert = FizzBuzzConverter.convert(150000);
        Assert.assertEquals("FizzBuzz", convert[convert.length - 1]);
    }
}

File: python/fizz_buzz/fizz_buzz_test.py
from nose.tools import istest, eq_
from fizz_buzz import fizz_buzz

@istest
def should_return_the_first_3_elements():
    eq_([1, 2, 'Fizz'], fizz_buzz(3))

@istest
def should_return_the_first_6_elements():
    eq_([1, 2, 'Fizz', 4, 'Buzz', 'Fizz'], fizz_buzz(6))


@istest
def should_return_fizz_buzz_for_15():
    result = fizz_buzz(15)
    eq_(15, len(result))
    eq_('FizzBuzz', result[-1])


@istest
def should_return_150000_elements():
    result = fizz_buzz(150000)
    eq_(150000, len(result))
    eq_('FizzBuzz', result[-1])
